apps/src/lib/kits/maker/ui/SetupChecklist.jsx

```diff
@@ -333,6 +361,35 @@ export default class SetupChecklist extends Component {
             )}
             {this.contactSupport()}
           </ValidationStep>
+          {experiments.isEnabled('microbit') &&
+            this.state.boardTypeDetected === BOARD_TYPE.MICROBIT &&
+            this.state[STATUS_BOARD_UPDATE_FIRMATA] !== Status.WAITING && (
```

Comment:
I think you could use another constant here, like
```
const firmataUpdateAttempted = expriments.isEnabled('microbit') && this.state.boardTypeDetected === BOARD_TYPE.MICROBIT && this.state[STATUS_BOARD_UPDATE_FIRMATA] !== Status.WAITING;
```

Sometimes sub-functions can be useful in situations like this too, like:
```
renderFirmataUpdateStep() {
  const boardUpdateStatus = this.state[STATUS_BOARD_UPDATE_FIRMATA];
  if (!experiments.isEnabled('microbit') || this.state.boardTypeDetected !== BOARD_TYPE.MICROBIT || boardUpdateStatus === Status.WAITING) {
    return;
  }

  let additionalDetails = null;
  if (boardUpdateStatus === Status.ATTEMPTING) {
    additionalDetails = <div>...</div>;
  } else if (boardUpdateStatus === Status.SUCCEEDED) {
    additionalDetails = <div>...</div>;
  }


  return <ValidationStep ...>{additionalDetails}</ValidationStep>;
}
```

I'm not 100% happy with the `additionalDetails` piece here. But generally, I like to keep the main `render` method pretty clean and extract business logic when I can.

Reply:
Thanks! I added 2 constants for now - `showUpdateMicroBitButton` and  `showUpdateMicroBitValidationStep`.

Reply:
https://codedotorg.atlassian.net/browse/SL-571